iot_verifier/src/gateway_cache.rs

```diff
@@ -1,93 +1,37 @@
-use futures::stream::StreamExt;
+use crate::gateway_updater::MessageReceiver;
 use helium_crypto::PublicKeyBinary;
-use iot_config::{
-    client::{Client as IotConfigClient, ClientError as IotConfigClientError},
-    gateway_info::{GatewayInfo, GatewayInfoResolver},
-};
-use retainer::Cache;
-use std::{sync::Arc, time::Duration};
-use tokio::task::JoinHandle;
-
-/// how long each cached items takes to expire ( 12 hours in seconds)
-const CACHE_TTL: Duration = Duration::from_secs(12 * (60 * 60));
-/// how often to evict expired items from the cache ( every 1 hour)
-const CACHE_EVICTION_FREQUENCY: Duration = Duration::from_secs(60 * 60);
+use iot_config::gateway_info::GatewayInfo;
 
 pub struct GatewayCache {
-    pub iot_config_client: IotConfigClient,
-    pub cache: Arc<Cache<PublicKeyBinary, GatewayInfo>>,
-    pub cache_monitor: JoinHandle<()>,
+    gateway_cache_receiver: MessageReceiver,
 }
 
 #[derive(Debug, thiserror::Error)]
 pub enum GatewayCacheError {
```

Comment:
do we think there are any other valid errors that can come from this cache? if not this could just be an error struct instead of an enum and i believe you can even define it in-line with the gateway cache struct so the result coming from `resolve_gateway_info` is just `Result<GatewayInfo, Self::Error>`

Reply:
At this point GatewayNotFound is the only error.  The enum itself is referenced in multiple modules and I think for maintainability ( should other errors be appended later ) keeping the enum is a better solution